### PR TITLE
PBI-23: See Lesson Material

### LINKED ISSUE #64: PBI-23: Teacher & Student can see lesson material in class
**Story points: 8**

- [x] Redirect to new page

### PR DESCRIPTION
**Pull request by fxdros**
This branch is kinda mixed up with PBI-22. Here's the detail:
1. Fix bug and complete wiring buttons on the `create lesson material` page
2. Create tests and implement user checking. Normal user (student) cannot access the `create` button
3. Mapping all lesson materials on each topic
4. Wiring classwork home page with the lesson material detail page
5. Add a 404 page to limit student's access

Screen Shot 2022-04-22 at 07 15 43

Screen Shot 2022-04-22 at 07 15 50

Screen Shot 2022-04-22 at 07 16 09

Screen Shot 2022-04-22 at 08 25 26

Screen Shot 2022-04-22 at 19 00 59

### CODE REVIEW

File: src/resources/app/pages/ClassworkPage.vue [hunk 1]
```diff
@@ -102,16 +159,44 @@ export default {
       try {
         const response = await getTopicAPI(courseId);
         this.topics = response.data;
-        for (let i=0; i<topics.length; i++) {
-          topics[i].push({courseId: this.$props.id})
+        console.log(response.data);
```
Comment (Zsaschz) [inline src/resources/app/pages/ClassworkPage.vue]: please remove log

File: src/resources/app/pages/ClassworkPage.vue [hunk 2]
```diff
@@ -102,16 +159,44 @@ export default {
...
+      } catch (error) {
+        await this.$store.dispatch("OPEN_SNACKBAR", "Error getting data");
+      }
+    },
+
+    isCurrentUserTheTeacher(teacher) {
+      this.userIsTeacher = this.$store.state.user.id === teacher;
+    },
+
+    dateSlicing(givenDate) {
```
Comment (Zsaschz) [inline src/resources/app/pages/ClassworkPage.vue]: i think we can isolate this to a separate utils class

File: src/resources/app/pages/course/CourseMaterial.vue [hunk 3]
```diff
@@ -30,17 +31,23 @@ export default {
...
+  methods: {
+    async getCourseMaterial() {
+      try {
+        let response = await getCourseMaterialById(this.$props.materialId);
+        if (response.data.topic === null){
+          response.data.topic = {title: ''}
+        }
+        this.material = response.data;
+      } catch (error) {
+      }
```
Comment (Zsaschz) [inline src/resources/app/pages/course/CourseMaterial.vue:47]: open snackbar

File: src/resources/app/pages/course/CreateCourseMaterial.vue [hunk 4]
```diff
@@ -122,10 +145,27 @@ export default {
...
-        this.$store.dispatch("OPEN_SNACKBAR", "Error uploading data");
+        await this.$store.dispatch("OPEN_SNACKBAR", "Error uploading data");
       }
       this.isSubmitting = false;
     },
+    async checkTeacher(courseId) {
+      try {
+        const response = await courseDetail(courseId);
+        if (response.data.createdBy !== this.$store.state.user.id){
+          window.location.assign("/404");
```
Comment (Zsaschz) [inline src/resources/app/pages/course/CreateCourseMaterial.vue]: using `window.location.assign()` will force the page to refresh, use `this.$inertia.visit()` instead for better user experience

**Review by Zsaschz — APPROVED** [review]
LGTM! Thanks for the revision and your hard work uwu

**Review by IlmaMannix — APPROVED** [review]
LGTM 👍

**Review by kanziebub — APPROVED** [review]

**Comment by kanziebub** [comment]
LGTM you did great uwu

**Pull request merged**